### PR TITLE
Analytics error tracking key/value

### PR DESCRIPTION
**Pull request by fredeil**
## Description

Add support for string overloads on TrackEvent and TrackError. Decided to not go for a valueTuple since it introduces an unnecessary dependency.

## Related PRs or issues

https://github.com/microsoft/appcenter-sdk-dotnet/issues/821

### CODE REVIEW

**Comment by fredeil** [comment]
Could someone give me details on the build?

cc @jwallra

File: SDK/AppCenter/Microsoft.AppCenter/Microsoft.AppCenter.csproj [hunk 1]
```diff
@@ -15,6 +15,7 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="MSBuild.Sdk.Extras" Version="1.1.0" PrivateAssets="All" />
+    <PackageReference Include="System.ValueTuple" Version="4.5.0" />
```
Comment (guperrot) [inline SDK/AppCenter/Microsoft.AppCenter/Microsoft.AppCenter.csproj]: This new dependency needs to be discussed.

**Comment by jwallra** [comment]
@fredeil I wasn't sure either, it seemed like it should be included in .NET, but it appears that it is only included by default after .NET 4.7, .NET Standard 2.0 and .NET Core 2.0.  See this stack overflow question: https://stackoverflow.com/questions/38382971/predefined-type-system-valuetuple%C2%B42%C2%B4-is-not-defined-or-imported

Since it is an external dependency we need to determine whether we want to take it or not

**Comment by fredeil** [comment]
Ok, maybe we could satisfy #821 by providing a string/string overloqd, like:

«void TrackEvent(string name, string key, string value)»

File: SDK/AppCenter/Microsoft.AppCenter/Microsoft.AppCenter.csproj (hunk 1 above)
Comment (guperrot) [reply SDK/AppCenter/Microsoft.AppCenter/Microsoft.AppCenter.csproj]: Also that's the cause of the build failure. It's not referenced correctly everywhere it's needed.

**Comment by guperrot** [comment]
> Ok, maybe we could satisfy #821 by providing a string/string overloqd, like:
> 
> «void TrackEvent(string name, string key, string value)»

Sure you can do that solution instead of the tuples.

**Comment by fredeil** [comment]
The tuples are removed now, System.ValueTuple is also gone now. Waiting for build results @jwallra @guperrot

**Comment by jwallra** [comment]
We're looking at this change now.  We're considering if we were to accept it would we need to implement something similar on other SDKs for consistency.

**Comment by guperrot** [comment]
The consistency with all other SDKs is being discussed.
We will also have a consistency issue with `Crashes.TrackError` and in other SDKs other APIs that take a dictionary.

**Comment by fredeil** [comment]
Ok, let me know what the outcome is

**Comment by guperrot** [comment]
Sorry that I didn't came back here earlier.
There is no plan for now to modify the signature on all SDKs.
Since the PR is linked to the feature request issue (which remains open), it won't be lost, so I can close here for now.

**Pull request closed without merging**